src/Tmds.MDns/NetworkInterfaceHandler.cs

```diff
@@ -453,6 +453,11 @@ private void RemoveService(Name name)
             if (service != null)
             {
                 _serviceInfos.Remove(name);
+                if (!_serviceHandlers.ContainsKey(name.SubName(1)))
+                {
+                    return;
```

Comment:
The service should not have been in _serviceInfos if there is no _serviceHandler for that name.
How did the service get in the _serviceInfos?